data/collector.go

```diff
@@ -373,7 +378,26 @@ func InvalidInterfaceErr(api resource.API) error {
 	return errors.Errorf("passed interface does not conform to expected resource type %s", api)
 }
 
+// NewFailedToReadErr constructs a new FailedToReadErr.
+func NewFailedToReadErr(component, method string, err error) error {
```

Comment:
This is now inconsistent with data.InvalidInterfaceErr - LMK what you think. I can change this back to FailedToReadErr and call the FailedToReadErr type something else

Reply:
I think you've got some lint errors: https://github.com/viamrobotics/rdk/actions/runs/14780595328/job/41498474408?pr=4961

Reply:
Thanks, fixed. This is ready for another look